src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -321,6 +325,56 @@ export const handleGetPublicForm: ControllerHandler<
           }
           return res.json({ form: publicForm, isIntranetUser })
         })
+    case FormAuthType.SGID_MyInfo: {
+      const accessTokenCookie = req.cookies[SGID_COOKIE_NAME]
+      if (!accessTokenCookie) {
+        return res.json({
+          form: publicForm,
+          isIntranetUser,
+        })
+      }
+      res.clearCookie(SGID_COOKIE_NAME)
+      res.clearCookie(MYINFO_LOGIN_COOKIE_NAME)
```

Comment:
If we are having separate auth type for `SGID_MyInfo`:

```
this.authType !== FormAuthType.MyInfo &&
      this.authType !== FormAuthType.SGID_MyInfo
```

why do we clear cookie for `SGID` and `MYINFO_LOGIN`; will there be a case where user has valid ongoing sessions for each of `SGID`, `MYINFO` and `SGID_MYINFO`?

Reply:
The cookie management is kind of confusing. I should probably make a third cookie type. Here's what each cookie is for:
1. `MYINFO_LOGIN_COOKIE_NAME` - contains singpass username (NRIC).
2. `SGID_COOKIE_NAME` - dual use. When using SGID Auth mode, contains the singpass username (NRIC), when using SGID MyInfo mode, is only a transient cookie holding the sgID access token.

When SGID MyInfo mode is used, we need to also set the MyInfo cookie as we need somewhere to store the singpass username, and also because we reuse MyInfo functions, which read and depend on that cookie.

I should probably split up the `SGID_COOKIE_NAME` into two SGID cookies for each SGID usage so that it's clearer, rather than reusing the same one. What do you think?

Reply:
Thanks for reply! I agree splitting `SGID_COOKIE_NAME` makes sense. Dual use cookie may not be ideal because `FormAuthType.SGID` sessions should be handled separately from `FormAuthType.SGID_MyInfo` (dual use implies starting one use case override the other use case)

Reply:
Thanks. I didn't think about that. That's a great observation!